enterprise_subsidy/apps/subsidy/models.py

```diff
@@ -327,3 +345,105 @@ def _create_redemption(self, learner_id, content_key, **kwargs):
+
+
+class SubsidyAccessPolicy(TimeStampedModelWithUuid):
+    """
+    (group, subsidy, catalog, access method, and optional total value)
+    """
+    class Meta:
+        abstract = True
+
+    group_uuid = models.UUIDField(null=True, blank=True, db_index=True)
+    catalog_uuid = models.UUIDField(null=True, blank=True, db_index=True)
+    access_method = AccessMethods.DIRECT
+
+    @property
+    def group_client(self):
+        return MOCK_GROUP_CLIENT
+
+    @classmethod
+    def get_policies_for_groups(cls, group_uuids):
+        return cls.objects.filter(group_uuid__in=group_uuids)
+
+    @property
+    def catalog_client(self):
+        return MOCK_CATALOG_CLIENT
+
+    def is_entitled(self, learner_id, content_key):
+        if not self.catalog_client.catalog_contains_content(self.catalog_uuid, content_key):
+            return False
+
+        if not self.group_client.group_contains_learner(self.group_uuid, learner_id):
+            return False
+
+        elif self.subsidy.is_redeemable(learner_id, content_key):
+            return True
+
+        return False
+
+    def use_entitlement(self, learner_id, content_key):
+        if self.is_entitled(learner_id, content_key):
+            if self.access_method == AccessMethods.DIRECT:
+                return self.subsidy.redeem(learner_id, content_key)
+            if self.access_method == AccessMethods.REQUEST:
+                self.subsidy.request_redemption(learner_id, content_key)
+        return False
+
+    def has_used_entitlement(self, learner_id, content_key):
+        if self.access_method == AccessMethods.DIRECT:
+            return self.subsidy.has_redeemed(learner_id, content_key)
+        if self.access_method == AccessMethods.REQUEST:
+            return self.subsidy.has_requested(learner_id, content_key)
+
+
+class SubscriptionAccessPolicy(SubsidyAccessPolicy):
+    """
+    """
+    subsidy = models.ForeignKey(SubscriptionSubsidy, null=True, on_delete=models.SET_NULL)
+
+
+class LearnerCreditAccessPolicy(SubsidyAccessPolicy):
+    """
+    """
+    subsidy = models.ForeignKey(LearnerCreditSubsidy, null=True, on_delete=models.SET_NULL)
+
+
+class PerLearnerEnrollmentCapLearnerCreditAccessPolicy(LearnerCreditAccessPolicy):
+    """
+    Example policy that limits the number of enrollments (really) transactions
+    that a learner is entitled to in a subsidy.
+    """
+    per_learner_cap = models.IntegerField(
+        blank=True,
+        default=0,
+    )
+    history = HistoricalRecords()
+
+    def is_entitled(self, learner_id, content_key):
+        with transaction.atomic():
+            if self.subsidy.transactions_for_learner(learner_id).count() < self.per_learner_cap:
+                return super().is_entitled(learner_id, content_key)
+            else:
+                return False
+
+
+class LicenseRequestAccessPolicy(SubscriptionAccessPolicy):
+    """
+    """
+    access_method = AccessMethods.REQUEST
+    history = HistoricalRecords()
```

Comment:
Delete all of this, it lives over in the subsidy access policy app in the enterprise-access repo.